db/background_mgr.go

```
@@ -406,13 +429,23 @@ func (b *BackgroundManager) UpdateHeartbeatDocClusterAware() error {
 		// If we've hit an error, and we haven't had a successful heartbeat in just under its TTL then we need to quit
 		// out. If we fail to write heartbeat for this time we can no longer ensure that this would be the only process
 		// running and another could end up starting.
-		if time.Now().Sub(b.clusterAwareOptions.lastSuccessfulHeartbeat) > (BackgroundManagerHeartbeatExpirySecs - BackgroundManagerHeartbeatIntervalSecs) {
+		if time.Now().Sub(time.Unix(b.clusterAwareOptions.lastSuccessfulHeartbeatUnix.Value(), 0)) > (BackgroundManagerHeartbeatExpirySecs - BackgroundManagerHeartbeatIntervalSecs) {
 			return err
 		}
 		return nil
 	}
 
-	b.clusterAwareOptions.lastSuccessfulHeartbeat = time.Now()
+	var status HeartbeatDoc
+	err = base.JSONUnmarshal(statusRaw, &status)
+	if err != nil {
+		return err
+	}
+
+	if status.ShouldStop {
+		_ = b.Stop()
```

Why is it safe to ignore error here?

---

Can just add logging here